src/api/l_graphics.c

```diff
@@ -1050,6 +1050,7 @@ static int l_lovrGraphicsNewShader(lua_State* L) {
           case LUA_TNUMBER: flag.id = lua_tointeger(L, -2); break;
           default: lovrThrow("Unexpected ShaderFlag key type (%s)", lua_typename(L, lua_type(L, -2)));
         }
+        lovrCheck(flags.length < UINT32_MAX, "Too many flags");
```

Comment:
Wouldn't `<=` be a more appropriate condition?  Also, the shader compiler only supports 2000 flags and LÖVR is currently reserving IDs 1000+, so maybe there could be a `MAX_FLAGS` constant since that's a more realistic limit.